test/test_ion_decimal.cpp

```diff
@@ -123,12 +123,15 @@
     ION_ASSERT_OK(ion_writer_write_ion_decimal(writer, &ion_decimal)); \
     ION_ASSERT_OK(ion_test_writer_get_bytes(writer, ion_stream, &result, &result_len));
 
+
 #define ION_DECIMAL_BINARY_READER_EXPECT_OVERFLOW(func, decimal_digits) \
     ION_DECIMAL_TEXT_TO_BINARY(decimal_digits); \
+    ION_ASSERT_OK(ion_reader_close(reader)); \
```

Comment:
`ION_DECIMAL_EXPECT_OVERFLOW` re-uses `reader`, allocating a new struct which leaks the original reader that was initialized in `ION_DECIMAL_TEXT_TO_BINARY`.